include/net/net_pkt.h

```diff
@@ -128,6 +128,11 @@ struct net_pkt {
 	u8_t ieee802154_rssi; /* Received Signal Strength Indication */
 	u8_t ieee802154_lqi;  /* Link Quality Indicator */
 #endif
+
+#if defined(CONFIG_NET_VLAN)
+	u16_t vlan_tag;
+#endif /* CONFIG_NET_VLAN */
```

Comment:
Why net_pkt would need to store that info?

Reply:
I noticed that iface variable that is feed to L2 ethernet.c is not always the proper value for a specific VLAN tag. Having this value set in net_pkt is also useful as figuring it out from iface is time consuming as iface does not actually know anything about VLANs.

Reply:
ok